tests/src/Simple/PInvoke/PInvoke.cs

```diff
@@ -436,6 +440,13 @@ private static void TestLastError()
             ThrowIfNotEquals(12345, Marshal.GetLastWin32Error(), "Last Error test failed");
         }
 
+        private static void TestHandleRef()
+        {
+            Console.WriteLine("Testing marshalling HandleRef");
+
+            ThrowIfNotEquals(true, HandleRefTest(new HandleRef(new object(), (IntPtr)2018), 2018), "HandleRef marshalling failed");
```

Comment:
It might be good to test the GC behavior since that's the whole point of HandleRef. You could do something like have the native method call up into managed, which triggers a GC.Collect.

Reply:
Tests like that are very fragile because codegen is free to extend lifetimes for various reasons - https://github.com/dotnet/corert/issues/2608#issuecomment-282168381 - if this was a new feature, I would write the test with a bunch of permutations to make it more likely we didn't collect just because codegen decided to extend lifetime anyway. But the value is pretty marginal. Hopefully there's coverage elsewhere in FX/CLR (I can't find it though, sigh).

I validated this by inspecting the generated IL.